dnsaas/admin.py

```diff
@@ -0,0 +1,258 @@
+        }),
+        (None, {'fields': ('created', 'modified')})
+    )
+    formfield_overrides = {
+        models.NullBooleanField: {
+            'widget': NullBooleanRadioSelect(
+                attrs={'class': 'radiolist inline'}
+            ),
+        },
+    }
+
+    def get_form(self, request, *args, **kwargs):
+        form = super().get_form(request, *args, **kwargs)
+        form.user = request.user
+        return form
+
+
+class RecordTemplateAdmin(ForeignKeyAutocompleteAdmin):
+    form = RecordAdminForm
+    list_display = RECORD_LIST_FIELDS
+
+
+class SuperMasterAdmin(admin.ModelAdmin):
+    list_display = ('ip', 'nameserver', 'account',)
+    list_filter = ('ip', 'account', 'created', 'modified')
+    search_fields = ('ip', 'nameserver',)
+    readonly_fields = ('created', 'modified')
+
+
+class DomainMetadataAdmin(ForeignKeyAutocompleteAdmin):
+    list_display = ('domain', 'kind', 'content',)
+    list_filter = ('kind', 'domain', 'created', 'modified')
+    list_per_page = 250
+    list_filter = ('created', 'modified')
+    readonly_fields = ('created', 'modified')
+    related_search_fields = {
+        'domain': ('name',),
+    }
+    save_on_top = True
+    search_fields = ('content',)
+
+
+class CryptoKeyAdmin(ForeignKeyAutocompleteAdmin):
+    list_display = ('domain', 'flags', 'active', 'content',)
+    list_filter = ('active', 'domain', 'created', 'modified')
+    list_per_page = 250
+    readonly_fields = ('created', 'modified')
+    related_search_fields = {
+        'domain': ('name',),
+    }
+    save_on_top = True
+    search_fields = ('content',)
+    formfield_overrides = {
+        models.NullBooleanField: {
+            'widget': NullBooleanRadioSelect(
+                attrs={'class': 'radiolist inline'}
+            ),
+        },
+    }
+
+
+class DeleteRequestAdmin(admin.ModelAdmin):
+    fields = ['owner', 'target_id', 'content_type']
+
+    def add_view(self, request, extra_context=None):
+        extra_context = extra_context or {}
+        extra_context['deleted_item'] = str(
+            ContentType.objects.get(
+                pk=request.GET['content_type']
+            ).get_object_for_this_type(pk=request.GET['target_id'])
+        )
+        return super().add_view(request, extra_context=extra_context)
+
+
+class RecordTemplateInline(admin.StackedInline):
+    model = RecordTemplate
+    extra = 1
+
+
+class DomainTemplateAdmin(ForeignKeyAutocompleteAdmin):
+    inlines = [RecordTemplateInline]
+    list_display = ['name', 'add_domain_link', 'is_public_domain']
+
+
+class DomainRequestAdmin(admin.ModelAdmin):
+    list_display = ['domain']
+    readonly_fields = ['key']
+
+
+class RecordRequestAdmin(admin.ModelAdmin):
+    list_display = ['target_' + field for field in RECORD_LIST_FIELDS]
+
+
+#TODO:: restrict all only for superuser
+#TODO:: requests readonly
+#TODO:: other admins
+#TODO:: fk from autocomplete
+admin_site2.register(Domain, DomainAdmin)
```

Comment:
couldn't we use register as a decorator, with site passed? https://docs.djangoproject.com/en/1.10/ref/contrib/admin/#django.contrib.admin.register